src/hivpy/resistance_mutations.py

```diff
@@ -727,6 +752,37 @@ def calc_pr88m_outcomes(self, on_taz, size):
 
         return p88_mutations
 
+    def calc_inm_outcomes(self, on_dol, on_cab, in_cab_tail, in_primary_infection, size):
+        """
+        Returns integrase inhibitor IN118, IN140, IN148, IN155, or IN263 majority mutation outcomes.
+        """
+        # outcomes on dol
+        prob_mutation = self.resist_rate_dol if on_dol else 0
+        in_mutations = rng.uniform(size=size) < prob_mutation
+
+        # outcomes on cab
+        prob_mutation = self.resist_rate_dol * self.cab_resist_factor if on_cab or in_cab_tail else 0
+        r = rng.uniform(size=size)
+        # increased risk during primary infection
+        if in_primary_infection:
+            r /= self.risk_change_cab_resist
+        in_mutations |= r < prob_mutation
+
+        return in_mutations
+
+    def calc_ca66m_outcomes(self, on_ole, on_len, in_len_tail, in_len_post_tail, on_art, size):
+        """
+        Returns CA gene CA66 mutation outcomes.
+        """
+        prob_mutation = self.resist_rate_len if on_ole or on_len or in_len_tail \
+            or (in_len_post_tail and not on_art) else 0
```

Comment:
Probably better to put this in a sub-pop statement before the function rather than splitting up a transform group on all these variables

Reply:
Let's look into the performance impact a bit

Reply:
So I created two slightly modified versions of `test_ca66m` and ran them with a population size of 100 000 (`update_new_mutations_arising_art` was called twice in each test). In one test everybody was infected with HIV, and in the other 5% of people were infected with HIV. I tested the original CA66 outcome calculation method, as well as a sub-pop method where I first selected all people either on oral Len, injectable Len, or in a Len tail, and then used that sub-pop with a modified CA66 calculation that featured fewer transform group columns.

Resulting average times from 5 runs of each test:
- `original_5%` = 3.5s
- `sub_pop_5%` = 3.4s
- `original_100%` = 22s
- `sub_pop_100%` = 23s

There doesn't seem to be much of a difference between the two methods.

Reply:
Look good, we can keep things as they are then!